.gitlab-ci.yml

```diff
@@ -21,20 +21,35 @@ variables:
   NFS_PATH: ${HL_NFS_OPPATH_CODON}
 
 stages:
+  - setup
   - test
   - build
   - deploy
 
-.deploy-rules:
+# Templates
+.branch-rules-main:
   rules:
     - if: '$CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "main"'
-      when: always
+      when: on_success
+    - when: never
+
+.branch-rules-dev:
+  rules:
+    - if: '$CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "dev"'
+      when: on_success
+    - when: never
 
-.deploy-branch-rules:
+.branch-rules-review:
   rules:
     - if: '$CI_DEPLOY_FREEZE == null && ($CI_COMMIT_BRANCH != "main" || $CI_COMMIT_BRANCH != "dev")'
```

Comment:
is `CI_DEPLOY_FREEZE` relevant for review deployments?

Reply:
To which env should deploy freeze mode apply to @kamaldodiya?

Reply:
It is not environment specific. if DEPLOY_FREEZE is set it would not allow that job to be scheduled - here deploy job to whatever (review) environment .

Reply:
@kamaldodiya @azangru so we should follow DEPLOY_FREEZE in all environment deployment jobs, including review. No changes needed here.

Reply:
My understanding was that `DEPLOY_FREEZE` only makes sense for us for production, or any other "important" deployment that for some reason should be prevented from updating immediately. Dev is on its way out; and feature review deployment (as well as probably staging?) should be updated immediately.

Reply:
Removed freeze check from review/staging.